### PR TITLE
Parse nested maps from viper configuration

### LINKED ISSUE #638: Fix broken parsing of HCL with maps
The standalone command-line tool is unable to properly parse map fields like `environment` because the code assumed that all the options can be represented as a single string.

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Closes #638; convoluted workaround to keep `cobra` instead of `viper` as the primary schema.

### CODE REVIEW

**Comment by dacbd** [comment]
Can you help my brain and give an example of what was broken and now fixed by this? I don't quite follow the info in the linked issue? 

this is broken?
```hcl
environment = { THINIG1 = "", THING2 = "" }
```
but this isn't?
```hcl
environment = { THINIG1 = "" }
```

**Comment by 0x2b3bfa0** [comment]
Both were broken. The standalone command-line tool ignored all the map-like attributes from a `main.tf` file.

**Review by dacbd — CHANGES_REQUESTED** [review]
It is improperly assigning values.

with:
```hcl
terraform {
  required_providers {
    iterative = { source = "github.com/iterative/iterative" }
  }
}
provider "iterative" {}
resource "iterative_task" "test" {
  cloud = "aws"
  machine = "t3.small"
  disk_size =-1
  storage {
    workdir = ""
    output = ""
  }
  environment = { 
    GITHUB_PAT= "",
    THING1 = "one", 
    THING2 = "two", 
    THING_ONE = "three",
    THING_TWO = "four"
  }
  script = <<-END
    #!/bin/bash
    env
  END
}
```

running the cli results with:
```
THING_TWO=two
GITHUB_PAT=*** #correctly set
THING_ONE=two
THING1=two
THING2=two
```

File: cmd/create/create.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func New(cloud *common.Cloud) *cobra.Command {
 func (o *Options) Run(cmd *cobra.Command, args []string, cloud *common.Cloud) error {
 	variables := make(map[string]*string)
 	for name, value := range o.Environment {
+		name = strings.ToUpper(name)
 		variables[name] = nil
 		if value != "" {
 			variables[name] = &value
```
Comment (0x2b3bfa0) [inline cmd/create/create.go:68]:
```suggestion
		if copy := value; value != "" {
			variables[name] = &copy
```

The issue you describe wasn't introduce by this pull request, but... 🤦🏼‍♂️

**Review by dacbd — APPROVED** [review]

**Pull request merged**